crates/hir-ty/src/lower.rs

```diff
@@ -191,10 +194,12 @@ impl<'db, 'a> TyLoweringContext<'db, 'a> {
     ) -> Self {
         let impl_trait_mode = ImplTraitLoweringState::new(ImplTraitLoweringMode::Disallowed);
         let in_binders = DebruijnIndex::ZERO;
+        let interner = DbInterner::new_with(db, resolver.krate(), None);
         Self {
             db,
-            interner: DbInterner::new_no_crate(db),
-            lang_items: hir_def::lang_item::lang_items(db, resolver.krate()),
+            // Can provide no block since we don't use it for trait solving.
```

Comment:
Didn't we use to use block impls in trait solving?

Reply:
This doesn't use trait solving, or even create an `InferCtxt`. We need the crate just for the lang items, because we use elaboration.

Reply:
Oh, I see. I misinterpreted the comment, the `it` in there I thought referred to the block not being used for trait solving. Not that we don't use the entire thing as there is no trait solving involved.